azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+	}
+
+	ctx := client.StopContext
+	_, error := managedClusterClient.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if error != nil {
+		return error
+	}
+
+	read, err := managedClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read AKS managed cluster %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for AKS managed cluster (%s) to become available", d.Get("name"))
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    ManagedClusterStateRefreshFunc(client, resGroup, name),
+		Timeout:    30 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for AKS managed cluster (%s) to become available: %s", d.Get("name"), err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmManagedClusterRead(d, meta)
+}
+
+func resourceArmManagedClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := meta.(*ArmClient).managedClustersClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["managedClusters"]
+
+	ctx := client.StopContext
+	resp, err := managedClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure Container Service %s: %s", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resGroup)
+	d.Set("dns_prefix", resp.DNSPrefix)
+	d.Set("kubernetes_version", resp.KubernetesVersion)
+
+	linuxProfile := flattenAzureRmManagedClusterLinuxProfile(*resp.ManagedClusterProperties.LinuxProfile)
+	d.Set("linux_profile", &linuxProfile)
+
+	agentPoolProfiles := flattenAzureRmManagedClusterAgentPoolProfiles(resp.ManagedClusterProperties.AgentPoolProfiles)
+	d.Set("agent_pool_profile", &agentPoolProfiles)
+
+	servicePrincipal := flattenAzureRmManagedClusterServicePrincipalProfile(resp.ManagedClusterProperties.ServicePrincipalProfile)
+	if servicePrincipal != nil {
+		d.Set("service_principal", servicePrincipal)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmManagedClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := client.managedClustersClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["managedClusters"]
+
+	ctx := client.StopContext
+	future, err := managedClusterClient.Delete(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error issuing Azure ARM delete request of Container Service '%s': %s", name, err)
```

Comment:
for the second formatting argument - can we make this `%+v` to return the full exception?

Reply:
Thanks, fixed.